src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -185,259 +189,192 @@ public void postSamlCallback(RoutingContext routingContext, Map<String, String>
-                              } else { //mod-authtoken v2.x returns 201, with token in JSON response body
-                                try {
-                                  candidateAuthToken = tokenResponse.getBody().getString("token");
-                                } catch(Exception e) {
-                                  asyncResultHandler.handle(Future.succeededFuture(PostSamlCallbackResponse.respond500WithTextPlain(e.getMessage())));
-                                }
-                              }
-                              final String authToken = candidateAuthToken;
-
-                              final String location = UriBuilder.fromUri(stripesBaseUrl)
-                                .path("sso-landing")
-                                .queryParam("ssoToken", authToken)
-                                .queryParam("fwd", originalUrl.getPath())
-                                .build()
-                                .toString();
-
-                              final String cookie = new NewCookie("ssoToken", authToken, "", originalUrl.getHost(), "", 3600, false).toString();
-
-                              HeadersFor302 headers302 = PostSamlCallbackResponse.headersFor302().withSetCookie(cookie).withXOkapiToken(authToken).withLocation(location);
-                              asyncResultHandler.handle(Future.succeededFuture(PostSamlCallbackResponse.respond302(headers302)));
-
-                            }
-                          });
-                      } catch (Exception httpClientEx) {
-                        asyncResultHandler.handle(Future.succeededFuture(PostSamlCallbackResponse.respond500WithTextPlain(httpClientEx.getMessage())));
-                      }
-                    }
+    findSaml2Client(routingContext, false, false, vertxContext)
+      .compose(samlClientComposite -> {
+        final SAML2Client client = samlClientComposite.getClient();
+        final SamlConfiguration configuration = samlClientComposite.getConfiguration();
+        String userPropertyName = configuration.getUserProperty() == null ? "externalSystemId" : configuration.getUserProperty();
+        String samlAttributeName = configuration.getSamlAttribute() == null ? "UserID" : configuration.getSamlAttribute();
 
-                  }
+        SAML2Credentials credentials = (SAML2Credentials) client.getCredentials(webContext, sessionStore).get();
 
+        // Get user id
+        List<?> samlAttributeList = (List<?>) credentials.getUserProfile().getAttribute(samlAttributeName);
+        if (samlAttributeList == null || samlAttributeList.isEmpty()) {
+          throw new UserErrorException("SAML attribute doesn't exist: " + samlAttributeName);
+        }
+        final String samlAttributeValue = samlAttributeList.get(0).toString();
+
+        final String usersCql = userPropertyName +
+          "=="
+          + QUOTATION_MARK_CHARACTER + samlAttributeValue + QUOTATION_MARK_CHARACTER;
```

Comment:
```suggestion
        final String usersCql = userPropertyName + "==" + StringUtil.cqlEncode(samlAttributeValue);
```
In addition userPropertyName should be checked against a white list.

Reply:
I'll do these things in #90.